### PR TITLE
Revert "inherit the async propagation from enclosing scope" and always propagate

### PR DESCRIPTION
**Pull request by tylerbenson**

### CODE REVIEW

**Comment by dougqh** [comment]
I don't understand the need to revert the inheritance change.  I think that change should be left in place until we remove async propagation entirely.

File: dd-java-agent/instrumentation/opentracing/api-0.31/src/test/groovy/OpenTracing31Test.groovy [hunk 1]
```diff
@@ -135,7 +135,7 @@ class OpenTracing31Test extends AgentTestRunner {
     expect:
     span instanceof MutableSpan
     scope instanceof TraceScope
-    !(scope as TraceScope).isAsyncPropagating()
+    (scope as TraceScope).isAsyncPropagating()
```
Comment (dougqh) [inline dd-java-agent/instrumentation/opentracing/api-0.31/src/test/groovy/OpenTracing31Test.groovy:138]: It seems strange to be checking this in OpenTracing & OpenTelemetry tests.
I'm not sure this assertion is adding much value.

File: dd-trace-core/src/main/java/datadog/trace/core/scopemanager/ContinuableScopeManager.java [hunk 2]
```diff
@@ -62,33 +59,17 @@ private ContinuableScopeManager(
...
-    return activate(span, source, true, isAsyncPropagating);
-  }
-
-  private AgentScope activate(
-      final AgentSpan span,
-      final ScopeSource source,
-      final boolean overrideAsyncPropagation,
-      final boolean isAsyncPropagating) {
-    ScopeStack scopeStack = scopeStack();
+    final ScopeStack scopeStack = scopeStack();
```
Comment (dougqh) [inline dd-trace-core/src/main/java/datadog/trace/core/scopemanager/ContinuableScopeManager.java:72]: I think this diff is from the inheritance revert.  As mentioned below, I don't think inheritance should be reverted as part of this change.

**Review by dougqh — CHANGES_REQUESTED** [review]
Reverting inheritance needs justification, I don't believe that needs to be part of this PR.
I actually believe reverting would be a significant step backwards.

**Comment by tylerbenson** [comment]
Closing for now.  We can revisit when we're ready to make more changes.

**Pull request closed without merging**